src/components/AnnotationViewer.vue

```diff
@@ -43,6 +44,27 @@ export default class AnnotationViewer extends Vue {
   readonly propertiesStore = propertiesStore;
   readonly filterStore = filterStore;
 
+  // Compute the centroid of each annotation, taking unrolling into account
+  get unrolledCentroidCoordinates() {
+    const centroids: { [annotationId: string]: IGeoJSPoint } = {};
+
+    const anyImage = this.store.dataset?.anyImage();
+    if (!anyImage) {
+      return centroids;
+    }
+
+    for (const annotation of this.annotationStore.annotations) {
+      const centroid = this.annotationStore.annotationCentroids[annotation.id];
+      const unrolledCentroid = this.unrolledCoordinates(
+        [centroid],
+        annotation.location,
+        anyImage
+      )[0];
+      centroids[annotation.id] = unrolledCentroid;
+    }
+    return centroids;
```

Comment:
```
const anyImage = this.store.dataset?.anyImage();
if (anyImage) {
  for (const annotation of this.annotationStore.annotations) {
    const centroid = this.annotationStore.annotationCentroids[annotation.id];
    const unrolledCentroid = this.unrolledCoordinates(
      [centroid],
      annotation.location,
      anyImage
    )[0];
    centroids[annotation.id] = unrolledCentroid;
  }
}
return centroids;
```

Reply:
Thanks, I forced push a commit to take your idea into account